ruby/tests/basic.rb

```diff
@@ -999,7 +999,6 @@ def test_to_h
         :optional_float=>0.0,
         :optional_int32=>0,
         :optional_int64=>0,
-        :optional_msg=>nil,
```

Comment:
why remove this line?

Reply:
This is because of the introduction of hasbit functionality, which mirrors the upb's implementation. A submessage is only considered to be present if not nil and the `to_h` method excludes everything which is not present.

If you want me to preserve the behavior for proto3, I can.

Reply:
http://ruby-doc.org/core-2.2.0/Array.html#method-i-to_h, from the doc, it seems that we only need to make sure same messages have same to_h result and different messages have different to_h result. So it has no restriction on whether we include unsetted fields.

I don't think the present behavior for message field has any change between proto2 and proto3. Message field always has presence check. So the to_h bevavior for message field should be consistent. I think not include unset message field in ```to_h``` is the right option.

@haberman, any thought? Can we still change this?

Reply:
I often look to Ruby's built-in `Struct` class to decide how things like this should behave. With Struct I notice:

```
> Struct.new("Foo", :a, :b).new.to_h
=> {:a=>nil, :b=>nil}
```

So I would be inclined to keep the nil values here. It matches the existing proto3 behavior which also seems right.

Reply:
I am fine preserving the proto3 behavior.

But for proto2, I will suggest to not include fields in `to_h` which aren't set. This is because proto2 uses `hasbit` for every field, including the basic types. So setting a value explicitly for a field in proto2, even for the default value, should set the `hasbit` for that field. Which means if `to_h` includes the field even when not set, it becomes impossible to know the difference between when the field was explicitly set and when it wasn't.